main.tf

```diff
@@ -9,23 +9,21 @@ terraform {
 provider "aws" {
   region = "us-west-2"
 }
-
-resource "aws_iam_role" "jay_project_role" {
-    name = "terraform_aws_lambda-role"
-    assume-role_policy = <<EOF
-  
-{
-    "Version": "2012-10-17",
-    "Statement".[
-        {
-            "Action": "sts:assumeRole,
-            "Principal": {
-                "Service": "lambda.amazonaws.com"
-            }
-            "Effect": "Allow",
-            "Sid": ""
-        },
-    ]
+data "aws_caller_identity" "current" {}
+locals {
+  account_id = data.aws_caller_identity.current.account_id
+  iam_role   = join(":", ["arn:iam:", local.account_id, "role/labRole"]) 
+}
+data "archive_file" "zip_the_python_code" {
+  type        =  "zip"
+  source_dir  =  "${path.module}/python/"
+  output_path =  "${path.module}/python/lambda.zip" 
```

Comment:
you put your zip in your source folder and zip it again

It would be better to have a build folder and also ignore this at git

```
data "archive_file" "zip_the_python_code" {
  type        =  "zip"
  source_dir  =  "${path.module}/python/"
  output_path =  "${path.module}/build/lambda.zip" 
}
```